### ISSUE TITLE
Possible race condition: NullPointerException: Attempt to invoke virtual MediaControllerCompat

### ISSUE THREAD

**Issue by bsutton** [dooboolab/flutter_sound#307]
I'm unable to reproduce the problem but lodging this incase it reoccurs and we have some additional information.

I produced this originally by:
Start example
Played MP3 Asset twice
Enabled Show on OS UI
Played MP3 asset twice

The following error was thrown:
```
W/MediaPlayer(26404): Use of stream types is deprecated for operations other than volume control
W/MediaPlayer(26404): See the documentation of setAudioStreamType() for what to use instead with android.media.AudioAttributes to qualify your playback use case
E/MethodChannel#com.dooboolab.flutter_sound_track_player(26404): Failed to handle method call
E/MethodChannel#com.dooboolab.flutter_sound_track_player(26404): java.lang.NullPointerException: Attempt to invoke virtual method 'android.support.v4.media.session.MediaControllerCompat$TransportControls android.support.v4.media.session.MediaControllerCompat.getTransportControls()' on a null object reference
E/MethodChannel#com.dooboolab.flutter_sound_track_player(26404): 	at com.dooboolab.fluttersound.TrackPlayer.startPlayerFromTrack(TrackPlayer.java:331)
E/MethodChannel#com.dooboolab.flutter_sound_track_player(26404): 	at com.dooboolab.fluttersound.TrackPlayerPlugin.onMethodCall(TrackPlayer.java:152)
E/MethodChannel#com.dooboolab.flutter_sound_track_player(26404): 	at io.flutter.plugin.common.MethodChannel$IncomingMethodCallHandler.onMessage(MethodChannel.java:231)
E/MethodChannel#com.dooboolab.flutter_sound_track_player(26404): 	at io.flutter.embedding.engine.dart.DartMessenger.handleMessageFromDart(DartMessenger.java:93)
E/MethodChannel#com.dooboolab.flutter_sound_track_player(26404): 	at io.flutter.embedding.engine.FlutterJNI.handlePlatformMessage(FlutterJNI.java:642)
E/MethodChannel#com.dooboolab.flutter_sound_track_player(26404): 	at android.os.MessageQueue.nativePollOnce(Native Method)
E/MethodChannel#com.dooboolab.flutter_sound_track_player(26404): 	at android.os.MessageQueue.next(MessageQueue.java:336)
E/MethodChannel#com.dooboolab.flutter_sound_track_player(26404): 	at android.os.Looper.loop(Looper.java:174)
E/MethodChannel#com.dooboolab.flutter_sound_track_player(26404): 	at android.app.ActivityThread.main(ActivityThread.java:7356)
E/MethodChannel#com.dooboolab.flutter_sound_track_player(26404): 	at java.lang.reflect.Method.invoke(Native Method)
```

Looking at the code the MediaControllerCompat is initialised by a onConnect callback.
I'm think that we tried to play media before the onConnect completed.

**Comment by bsutton** [comment]
I've had this occur again. Seems to only occur when I select Use OS Media Player.

**Comment by Larpoux** [comment]
Great !  Personally I was almost sure that the problem was caused by  the `onConnect` callback not finished when the App starts its playback.

I put a patch to avoid the crash in 3.x.x, but this should be cleaned in a better way.

**Comment by bsutton** [comment]
do you think the onConnect is failing or are we simply not waiting for it
to complete?

**Comment by Larpoux** [comment]
Not sure. When running on debugger the problem does not occur.
Not really easy to debug.

**Comment by bsutton** [comment]
I would suggest that we need to modify the code to stop the race condition.
I can't see what triggers the onConnect but it looks to me like there is no code in place to force startPlayerFromTrack to wait for the onConnect.

We need to put some sort of lock in place to for startPlayerFromTrack to wait.
I'm not certain what locking mechanisms are available on android?
In server side java I would normally use something like a countdownlatch.

**Comment by Larpoux** [comment]
What about this PR (Problem Report) ?
Is it still relevant in 4.2.2 ?
Are you sure to `await` initialize() before trying to playback your record ?

**Comment by Larpoux** [comment]
Fixed in 4.0.3+1.
Please reopen this issue if this Problem occurs again.
